### PR TITLE
qa: add iscsi smoke test

### PR DESCRIPTION
**Pull request by rjfd**
Signed-off-by: Ricardo Dias <rdias@suse.com>

### CODE REVIEW

**Comment by smithfarm** [comment]
@rjfd Please rebase this on top of `ses6-deepsea-pr` branch

**Comment by smithfarm** [comment]
This is missing a method in `qa/tasks/deepsea.py` Validation class.

**Comment by rjfd** [comment]
@smithfarm rebased and added the missing method

**Comment by smithfarm** [comment]
@rjfd The script completes successfully, but this part doesn't look right:

```
2019-03-27T11:35:54.984 INFO:teuthology.orchestra.run.target192168000004.stdout:My IP: 192.168.0.4
2019-03-27T11:35:54.984 INFO:teuthology.orchestra.run.target192168000004.stdout:Other gateways: 192.168.0.17
2019-03-27T11:35:54.984 INFO:teuthology.orchestra.run.target192168000004.stdout:RBD pool: iscsi-images
2019-03-27T11:35:56.681 INFO:teuthology.orchestra.run.target192168000004.stderr:++ curl --user admin:admin -X GET http://192.168.0.4:5000/api/sysinfo/hostname
2019-03-27T11:35:56.685 INFO:teuthology.orchestra.run.target192168000004.stderr:++ jq -r .data
2019-03-27T11:35:56.714 INFO:teuthology.orchestra.run.target192168000004.stderr:  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
2019-03-27T11:35:56.715 INFO:teuthology.orchestra.run.target192168000004.stderr:                                 Dload  Upload   Total   Spent    Left  Speed
2019-03-27T11:35:56.716 INFO:teuthology.orchestra.run.target192168000004.stderr:
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100    30  100    30    0     0  15000      0 --:--:-- --:--:-- --:--:-- 15000
2019-03-27T11:35:56.717 INFO:teuthology.orchestra.run.target192168000004.stderr:+ HOST=target192168000004
2019-03-27T11:35:56.717 INFO:teuthology.orchestra.run.target192168000004.stderr:+ gwcli /iscsi-targets/iqn.2003-01.com.redhat.iscsi-gw:iscsi-igw/gateways create target192168000004 192.168.0.4 skipchecks=true
2019-03-27T11:35:58.395 INFO:teuthology.orchestra.run.target192168000004.stderr:+ for ip in "${other_ips[@]}"
2019-03-27T11:35:58.396 INFO:teuthology.orchestra.run.target192168000004.stderr:++ curl --user admin:admin -X GET http://192.168.0.17:5000/api/sysinfo/hostname
2019-03-27T11:35:58.397 INFO:teuthology.orchestra.run.target192168000004.stderr:++ jq -r .data
2019-03-27T11:35:58.430 INFO:teuthology.orchestra.run.target192168000004.stderr:  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
2019-03-27T11:35:58.430 INFO:teuthology.orchestra.run.target192168000004.stderr:                                 Dload  Upload   Total   Spent    Left  Speed
2019-03-27T11:35:58.431 INFO:teuthology.orchestra.run.target192168000004.stderr:
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to 192.168.0.17 port 5000: Connection refused
2019-03-27T11:35:58.431 INFO:teuthology.orchestra.run.target192168000004.stderr:+ HOST=
2019-03-27T11:35:58.432 INFO:teuthology.orchestra.run.target192168000004.stderr:+ gwcli /iscsi-targets/iqn.2003-01.com.redhat.iscsi-gw:iscsi-igw/gateways create 192.168.0.17 skipchecks=true
2019-03-27T11:35:59.489 INFO:teuthology.orchestra.run.target192168000004.stderr:Missing required parameter ip_address
2019-03-27T11:35:59.529 INFO:teuthology.orchestra.run.target192168000004.stderr:+ for ip in "${other_ips[@]}"
2019-03-27T11:35:59.529 INFO:teuthology.orchestra.run.target192168000004.stderr:++ curl --user admin:admin -X GET http://192.168.0.3:5000/api/sysinfo/hostname
2019-03-27T11:35:59.529 INFO:teuthology.orchestra.run.target192168000004.stderr:++ jq -r .data
2019-03-27T11:35:59.564 INFO:teuthology.orchestra.run.target192168000004.stderr:  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
2019-03-27T11:35:59.564 INFO:teuthology.orchestra.run.target192168000004.stderr:                                 Dload  Upload   Total   Spent    Left  Speed
2019-03-27T11:35:59.564 INFO:teuthology.orchestra.run.target192168000004.stderr:
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to 192.168.0.3 port 5000: Connection refused
2019-03-27T11:35:59.564 INFO:teuthology.orchestra.run.target192168000004.stderr:+ HOST=
2019-03-27T11:35:59.565 INFO:teuthology.orchestra.run.target192168000004.stderr:+ gwcli /iscsi-targets/iqn.2003-01.com.redhat.iscsi-gw:iscsi-igw/gateways create 192.168.0.3 skipchecks=true
2019-03-27T11:36:00.657 INFO:teuthology.orchestra.run.target192168000004.stderr:Missing required parameter ip_address
2019-03-27T11:36:00.704 INFO:teuthology.orchestra.run.target192168000004.stderr:+ for ip in "${other_ips[@]}"
2019-03-27T11:36:00.704 INFO:teuthology.orchestra.run.target192168000004.stderr:++ curl --user admin:admin -X GET http://192.168.0.20:5000/api/sysinfo/hostname
2019-03-27T11:36:00.704 INFO:teuthology.orchestra.run.target192168000004.stderr:++ jq -r .data
2019-03-27T11:36:00.739 INFO:teuthology.orchestra.run.target192168000004.stderr:  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
2019-03-27T11:36:00.739 INFO:teuthology.orchestra.run.target192168000004.stderr:                                 Dload  Upload   Total   Spent    Left  Speed
2019-03-27T11:36:00.740 INFO:teuthology.orchestra.run.target192168000004.stderr:
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to 192.168.0.20 port 5000: Connection refused
2019-03-27T11:36:00.740 INFO:teuthology.orchestra.run.target192168000004.stderr:+ HOST=
2019-03-27T11:36:00.740 INFO:teuthology.orchestra.run.target192168000004.stderr:+ gwcli /iscsi-targets/iqn.2003-01.com.redhat.iscsi-gw:iscsi-igw/gateways create 192.168.0.20 skipchecks=true
2019-03-27T11:36:01.867 INFO:teuthology.orchestra.run.target192168000004.stderr:Missing required parameter ip_address
```

**Comment by smithfarm** [comment]
Also, the cluster configuration was historically increased from single-node to 4 nodes to accommodate the old lrbd-based test, which stopped working in a single-node cluster at some point.

Do you think this test would work well in a single-node configuration? (I.e., the whole shebang on a single node - makes the tests run faster)

**Comment by rjfd** [comment]
> Do you think this test would work well in a single-node configuration?

No, this test assumes that there are at least two iSCSI gateways

**Comment by rjfd** [comment]
@smithfarm I pushed a patch to the test script that will fix the curl errors shown in the above log. And I also changed the roles of the suite to include just two nodes and two iscsi gateways.

File: qa/suites/deepsea/tier2/igw/cluster/roles.yaml [hunk 1]
```diff
@@ -1,5 +1,4 @@
 roles:
-  - [client.salt_master, mon.a, mgr.x, osd.0]
-  - [mon.b, mgr.y, osd.0]
-  - [mon.c, mgr.z, osd.0]
-  - [igw.0, client.0]
+  - [client.salt_master, mon.a, mgr.x, osd.0, igw.0]
+  - [igw.1, client.0]
+
```
Comment (smithfarm) [inline qa/suites/deepsea/tier2/igw/cluster/roles.yaml]: please drop the concluding newline (probably harmless, but still)

Comment (rjfd) [reply qa/suites/deepsea/tier2/igw/cluster/roles.yaml]: done

**Pull request merged**